routers/private/hook.go

```diff
@@ -92,6 +105,154 @@ func HookPreReceive(ctx *macaron.Context, opts private.HookOptions) {
 
 				}
 			}
+
+			// Require signed commits
+			if protectBranch.RequireSignedCommits {
+				env := os.Environ()
```

Comment:
we should move the **set env codeblock** before Line 75 and remove 78-90 to remove dublicated code